crates/ide-completion/src/context/analysis.rs

```diff
@@ -1875,11 +1875,37 @@ fn is_in_breakable(node: &SyntaxNode) -> BreakableKind {
 }
 
 fn is_in_block(node: &SyntaxNode) -> bool {
+    if has_in_newline_expr_first(node) {
+        return true;
+    };
     node.parent()
         .map(|node| ast::ExprStmt::can_cast(node.kind()) || ast::StmtList::can_cast(node.kind()))
         .unwrap_or(false)
 }
 
+/// Similar to `has_parens`, heuristic sensing incomplete statement before ambigiguous `Expr`
+///
+/// Heuristic:
+///
+/// If the `PathExpr` is left part of the `Expr` and there is a newline after the `PathExpr`,
+/// it is considered that the `PathExpr` is not part of the `Expr`.
+fn has_in_newline_expr_first(node: &SyntaxNode) -> bool {
+    if node.kind() == SyntaxKind::PATH_EXPR
```

Comment:
```suggestion
    if ast::PathExpr::can_cast(node.kind())
```
I prefer the `can_cast()` methods as it's easy to get the wrong `SyntaxKind` accidentally.